src/stream/src/executor/source/fs_fetch_executor.rs

```diff
@@ -312,100 +336,85 @@ impl<S: StateStore, Src: OpendalSource> FsFetchExecutor<S, Src> {
                                 // store into state table.
                                 Message::Chunk(chunk) => {
                                     // For Parquet encoding, the offset indicates the current row being read.
-                                    // Therefore, to determine if the end of a Parquet file has been reached, we need to compare its offset with the total number of rows.
-                                    // We directly obtain the total row count and set the size in `OpendalFsSplit` to this value.
-                                    let file_assignment = if let EncodingProperties::Parquet =
-                                        source_desc.source.parser_config.encoding_config
-                                    {
-                                        let filename_list: Vec<_> = chunk
-                                            .data_chunk()
-                                            .rows()
-                                            .map(|row| {
-                                                let filename = row.datum_at(0).unwrap().into_utf8();
-                                                filename.to_owned()
-                                            })
-                                            .collect();
-                                        let mut parquet_file_assignment = vec![];
-                                        for filename in &filename_list {
-                                            let total_row_num =
-                                                get_total_row_nums_for_parquet_file(
-                                                    filename,
-                                                    source_desc.clone(),
-                                                )
-                                                .await?;
-                                            parquet_file_assignment.push(
-                                                OpendalFsSplit::<Src>::new(
-                                                    filename.to_owned(),
-                                                    0,
-                                                    total_row_num - 1, // -1 because offset start from 0.
-                                                ),
+                                    let file_assignment = chunk
+                                        .data_chunk()
+                                        .rows()
+                                        .map(|row| {
+                                            let filename = row.datum_at(0).unwrap().into_utf8();
+
+                                            let size = row.datum_at(2).unwrap().into_int64();
```

Comment:
prefer avoid hard cod

Reply:
+1 for avoiding hard code. 
This is just following the previous approach. Since the fetch executor is not aware of the schema, this is hard coded after the fetch executor is introduced, which is also the reason for the parquet include hidden_column panic I mentioned above.